newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrExpressionSelector/index.js

```diff
@@ -52,178 +52,167 @@ type Props<T> = {|
   onClickMore?: () => void,
   id?: ?string,
 |};
-type State<T> = {|
+type SearchState<T> = {|
   searchText: string,
   searchResults: Array<T>,
 |};
 
-export default class InstructionOrExpressionSelector<
+const InstructionOrExpressionSelector = <
   T: EnumeratedInstructionOrExpressionMetadata
-> extends React.PureComponent<Props<T>, State<T>> {
-  state: State<T> = {
+>({
+  focusOnMount,
+  instructionsInfo,
+  instructionsInfoTree,
+  selectedType,
+  onChoose,
+  iconSize,
+  useSubheaders,
+  searchPlaceholderObjectName,
+  searchPlaceholderIsCondition,
+  helpPagePath,
+  style,
+  onClickMore,
+  id,
+}: Props<T>) => {
+  const searchBarRef = React.useRef<?SearchBarInterface>(null);
+  const scrollViewRef = React.useRef<?ScrollViewInterface>(null);
+  const selectedItemRef = React.useRef<?ListItemRefType>(null);
+  const [searchState, setSearchState] = React.useState<SearchState<T>>({
```

Comment:
do you think it's better to use a common state rather than 2 different ones?
Is there a risk the results component re-renders only when the text change event if the results didn't?

Reply:
It looks like the searchResults are not even used?